TAM: Todas las PyMEs en México que venden productos perecederos, sin importar su tamaño, ubicación o capacidad tecnológica.

SAM (Mercado Accesible Servible): Aquellas PyMEs dentro del TAM que son accesibles para tu negocio, considerando factores como tamaño, ubicación, disposición a pagar y compatibilidad tecnológica.

SOM (Mercado Objetivo Servible): El segmento específico de PyMEs al que te dirigirás inicialmente, basado en tus recursos, estrategia y objetivos.

##TAM

In [ ]:
#Comenzaremos cargando nuestro archivo csv
import pandas as pd
Datos_Tam = pd.read_csv("/content/PYMES_LISTO.csv")
Datos_Tam.head()

Unnamed: 0              nom_estab  codigo_act  \
0           0    ABARR0TES DON PEDRO      461110   
1           1    ABARR0TES EL SAIDER      461110   
2           2      ABARR0TES PUENTES      461110   
3           3             ABARROTERA      461110   
4           4  ABARROTERA DEL CENTRO      461110   

                                          nombre_act          per_ocu  \
0  Comercio al por menor en tiendas de abarrotes,...   0 a 5 personas   
1  Comercio al por menor en tiendas de abarrotes,...   0 a 5 personas   
2  Comercio al por menor en tiendas de abarrotes,...   0 a 5 personas   
3  Comercio al por menor en tiendas de abarrotes,...  6 a 10 personas   
4  Comercio al por menor en tiendas de abarrotes,...   0 a 5 personas   

           Estado    latitud    longitud       fecha  per_ocu_num  \
0  Aguascalientes  21.902728 -102.247834  2019-11-01          2.0   
1  Aguascalientes  21.868223 -102.232901  2014-12-01          2.0   
2  Aguascalientes  22.248455 -102.333434  2014-12-01          2.0   
3  Aguascalientes  21.847281 -102.704958  2010-07-01          8.0   
4  Aguascalientes  22.366288 -102.297583  2019-11-01          2.0   

  clasificacion_empresa  
0          Microempresa  
1          Microempresa  
2          Microempresa  
3          Microempresa  
4          Microempresa

In [63]:
#Checaremos el valor maximo de per_ocu_num
Datos_Tam["per_ocu_num"].max()
#Checaremos cuantas empresas estan clasificadas como: No Pymes, en la columna clasificacion_empresas
Datos_Tam["clasificacion_empresa"].value_counts()
#Pasaremos a renombrar Todos los datos que estan clasificados como No es Pymes por Grandes Empresas
Datos_Tam['clasificacion_empresa'] = Datos_Tam['clasificacion_empresa'].replace('No es Pymes', 'Grandes Empresas')
Datos_Tam.count()

Unnamed: 0               30539
nom_estab                30539
codigo_act               30539
nombre_act               30539
per_ocu                  30539
Estado                   30539
latitud                  30539
longitud                 30539
fecha                    30539
per_ocu_num              30510
clasificacion_empresa    30539
dtype: int64

In [ ]:
#Pasaremos a descargar nuestros datos a un archivo de csv
Datos_Tam.to_csv('TAM.csv', index=False)

In [ ]:
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd

# Cargar el archivo CSV
df = pd.read_csv('/content/TAM.csv')
# Convertir 'fecha' a datetime, especificando el formato '%Y-%m'
df['fecha'] = pd.to_datetime(df['fecha'], format='%Y-%m', errors='coerce')
# The errors='coerce' argument will handle any invalid dates by setting them to NaT (Not a Time)

# 1. Tamaño del Mercado Geográfico
geo_group = df.groupby('Estado')['per_ocu'].sum().reset_index()
print("Tamaño del mercado por estado:")
print(geo_group)

# 2. Segmentación por Clasificación Empresarial
clasif_group = df.groupby('clasificacion_empresa')['per_ocu'].sum().reset_index()
print("\nDistribución de personal ocupado por clasificación empresarial:")
print(clasif_group)

# 3. Potencial de Mercado por Actividad Económica
actividad_group = df.groupby('nombre_act')['per_ocu'].sum().reset_index()
print("\nPotencial de mercado por actividad económica:")
print(actividad_group)

# 4. Análisis Temporal
# Group by month and sum 'per_ocu'
temporal_group = df.groupby(df['fecha'].dt.to_period('M'))['per_ocu'].sum().reset_index()
# Convert the 'fecha' column back to string for plotting
temporal_group['fecha'] = temporal_group['fecha'].astype(str)
print("\nEvolución del mercado a lo largo del tiempo:")
print(temporal_group)


Tamaño del mercado por estado:
                  Estado                                            per_ocu
0         Aguascalientes  0 a 5 personas0 a 5 personas0 a 5 personas6 a ...
1        Baja California  6 a 10 personas0 a 5 personas6 a 10 personas0 ...
2    Baja California Sur  11 a 30 personas11 a 30 personas31 a 50 person...
3               Campeche  0 a 5 personas0 a 5 personas0 a 5 personas11 a...
4                Chiapas  0 a 5 personas0 a 5 personas0 a 5 personas6 a ...
5              Chihuahua  11 a 30 personas0 a 5 personas0 a 5 personas10...
6   Coahuila de Zaragoza  0 a 5 personas0 a 5 personas0 a 5 personas6 a ...
7                 Colima  101 a 250 personas101 a 250 personas51 a 100 p...
8                 Puebla  0 a 5 personas0 a 5 personas0 a 5 personas0 a ...
9              Querétaro  0 a 5 personas0 a 5 personas0 a 5 personas0 a ...
10          Quintana Roo  0 a 5 personas0 a 5 personas31 a 50 personas0 ...

Distribución de personal ocupado por clasificación empre

##SAM

Micro: pueden ser de cualquier sector con hasta 10 trabajadores y un rango de ventas anuales de hasta 4 millones de pesos.

Pequeña: esta categoría es integrada por PYMES del sector de comercio, industrias y servicios, con hasta 30 trabajadores para el comercio y 50 para industrias y servicios. Además, el tope de ingresos anuales es de hasta 100 millones de pesos tanto para el comercio como para industrias y servicios.

Medianas: esta categoría está integrada por empresas de los sectores de comercios, servicios e industria. Comercio y servicios deben tener un máximo de hasta 100 trabajadores; industria debe tener un máximo de hasta 250 trabajadores. El tope de ingresos para comercio, servicios e industria es de hasta 250 millones de pesos al año.

In [64]:
#El siguiente paso es enfocarnos en el Mercado Accesible Servible
Datos_Tam['clasificacion_empresa'].value_counts() # Corrected the method name to value_counts()

Datos_Sam = Datos_Tam[Datos_Tam['clasificacion_empresa'].isin(['Grandes Empresas', 'Mediana empresa', 'Pequeña Empresa'])]
Datos_Sam.head()


Unnamed: 0               411
nom_estab                411
codigo_act               411
nombre_act               411
per_ocu                  411
Estado                   411
latitud                  411
longitud                 411
fecha                    411
per_ocu_num              382
clasificacion_empresa    411
dtype: int64

## OTRA IDEA

In [72]:
# Importar las bibliotecas necesarias
import pandas as pd
from pyspark.sql import SparkSession

# Inicializar la sesión de Spark
spark = SparkSession.builder.appName("Datos_Tam").getOrCreate()

# Cargar el archivo CSV en un DataFrame de pandas
Datos_Tam = pd.read_csv("/content/TAM.csv")

# Convertir el DataFrame de pandas a un DataFrame de Spark
df_spark = spark.createDataFrame(Datos_Tam)

# Convertir el DataFrame de Spark a un RDD
rdd = df_spark.rdd

# Definir los valores a eliminar
valores_a_eliminar = [
    "Comercio al por menor en tiendas de abarrotes, ultramarinos y misceláneas",
    "Cafeterías, fuentes de sodas, neverías, refresquerías y similares",
    "Servicios de preparación de otros alimentos para consumo inmediato",
    "Comercio al por menor de frutas y verduras frescas",
    "Comercio al por menor de carnes rojas",
    "Comercio al por menor en tiendas de abarrotes",
    "Bares, cantinas y similares",
    "Comercio al por menor de otros alimentos",
    "Comercio al por menor de paletas de hielo y helados",
    "Comercio al por menor de bebidas no alcohólicas y hielo",
    "Comercio al por menor de pescados y mariscos",
    "Comercio al por menor de carne de aves",
    "Comercio al por menor de leche, otros productos lácteos y embutidos",
    "Centros nocturnos, discotecas y similares",
    "Servicios de preparación de alimentos en unidades móviles"
]

# Filtrar el RDD eliminando las filas con los valores de 'nombre_act' en la lista 'valores_a_eliminar'
# Accedemos a la columna 'nombre_act' usando su nombre
rdd_filtrado = rdd.filter(lambda x: x['nombre_act'] not in valores_a_eliminar)

# Verificar los primeros elementos del RDD filtrado (solo las primeras 5 filas)
print(rdd_filtrado.take(5))  # Esto imprimirá las filas del RDD filtrado

# Convertir el RDD filtrado de nuevo a un DataFrame de Spark (si es necesario)
df_filtrado = spark.createDataFrame(rdd_filtrado, df_spark.columns)

# Verificar el DataFrame filtrado
df_filtrado.show()
df_filtrado.count()

[Row(Unnamed: 0=0, nom_estab='ABARR0TES DON PEDRO', codigo_act=461110, nombre_act='Comercio al por menor en tiendas de abarrotes, ultramarinos y miscel�neas', per_ocu='0 a 5 personas', Estado='Aguascalientes', latitud=21.9027281, longitud=-102.24783443, fecha='2019-11-01', per_ocu_num=2.0, clasificacion_empresa='Microempresa'), Row(Unnamed: 0=1, nom_estab='ABARR0TES EL SAIDER', codigo_act=461110, nombre_act='Comercio al por menor en tiendas de abarrotes, ultramarinos y miscel�neas', per_ocu='0 a 5 personas', Estado='Aguascalientes', latitud=21.86822328, longitud=-102.23290068, fecha='2014-12-01', per_ocu_num=2.0, clasificacion_empresa='Microempresa'), Row(Unnamed: 0=2, nom_estab='ABARR0TES PUENTES', codigo_act=461110, nombre_act='Comercio al por menor en tiendas de abarrotes, ultramarinos y miscel�neas', per_ocu='0 a 5 personas', Estado='Aguascalientes', latitud=22.24845496, longitud=-102.33343385, fecha='2014-12-01', per_ocu_num=2.0, clasificacion_empresa='Microempresa'), Row(Unnamed:

18106

In [74]:
#Aplicaremos otro filtro a nuestro df, ya que queremos que aprtir de la columna per_ocu nos muestre los valores a partir de 6 personas
# Filtrar el DataFrame para mostrar solo las filas donde per_ocu_num > 8
df_filtrado = df_spark.filter(df_spark.per_ocu_num >= 8)

# Mostrar el resultado
df_filtrado.show()
df_filtrado.count()

+----------+--------------------+----------+--------------------+---------------+--------------+-----------+-------------+----------+-----------+---------------------+
|Unnamed: 0|           nom_estab|codigo_act|          nombre_act|        per_ocu|        Estado|    latitud|     longitud|     fecha|per_ocu_num|clasificacion_empresa|
+----------+--------------------+----------+--------------------+---------------+--------------+-----------+-------------+----------+-----------+---------------------+
|         3|          ABARROTERA|    461110|Comercio al por m...|6 a 10 personas|Aguascalientes|21.84728125| -102.7049581|2010-07-01|        8.0|         Microempresa|
|        91|     ABARROTES ADAME|    461110|Comercio al por m...|6 a 10 personas|Aguascalientes|22.14761255|-102.41541554|2010-07-01|        8.0|         Microempresa|
|       150|      ABARROTES ALEX|    461110|Comercio al por m...|6 a 10 personas|Aguascalientes|21.93226265|-102.25514291|2010-07-01|        8.0|         Microe

4378

Con la aplicación de técnicas de Big Data, se identificó que, del total de empresas que realizaron actividades en nuestro país a través de Internet, 77.2 % utilizó su sitio web con fines únicamente informativos y publicitarios y 21.8 % lo utilizó para
ofrecer bienes y/o servicios en línea.

• Los principales medios de pago para las transacciones en Internet fueron las
plataformas de pagos electrónicos (45.7 %) y las tarjetas bancarias (30.7 %).

• De acuerdo con los Censos Económicos 2019, los negocios con presencia en Internet representaron 7.3 % del total de establecimientos, aportaron 41.1 % del personal ocupado y 57.6 % de los ingresos que se captaron.




La consultora SELECT, basada en la respuesta de directivos y especialistas del sector, estima que los proyectos más relevantes durante el 2024 se concentren en cuatro aristas: La aceleración en el desarrollo de productos y servicios de IA, nube y ciberseguridad; La implementación de estrategias de marketing digital para promover la venta en canales digitales; El control de los flujos de efectivos con mejores planes de presupuesto, y la consolidación de certificaciones avanzadas en nuevas tecnologías.

In [81]:
# Convert the PySpark DataFrame to a Pandas DataFrame
df_pandas = df_filtrado.toPandas()

# Now you can use to_csv on the Pandas DataFrame
df_pandas.to_csv('SAM1.csv', index=False)